File: beagle-declarative/src/main/kotlin/br/com/zup/beagle/declarative/widget/layout/PageView.kt
package br.com.zup.beagle.declarative.widget.layout

import br.com.zup.beagle.declarative.widget.core.Widget

interface PageIndicatorWidget : Widget

data class PageView(
    val pages: List<Widget>,
    val pageIndicator: PageIndicatorWidget? = null
) : Widget

File: Sources/BeagleUI/Sources/Widgets/Page/PageView.swift
//
//  Copyright © 21/11/19 Zup IT. All rights reserved.
//

import Foundation
import UIKit

public struct PageView: Widget {

    public let pages: [Widget]
    public let pageIndicator: PageIndicator?

    public init(
        pages: [Widget],
        pageIndicator: PageIndicator?
    ) {
        self.pages = pages
        self.pageIndicator = pageIndicator
    }
}

public protocol PageIndicator: Widget {}
